database/meta.go

```diff
@@ -95,27 +95,31 @@ func (m *MetaDB) SetRollingSettings(db string, chunk int, numchunks int) error {
 	ssn := m.dbHandle.Copy()
 	defer ssn.Close()
 
+	selector := bson.M{"name": db}
+	update := bson.M{
+		"rolling": true,
+		"current_chunk": chunk,
+		"total_chunks": numchunks,
+	}
+
 	if !result.Rolling {
 		m.log.Infof("The dataset [ %s ] is being converted to a rolling dataset.", db)
-	} else if result.TotalChunks != numchunks {
-		m.log.Warnf("The total chunk size for existing rolling dataset [ %s ] was set to [ %d ] and is being changed to [ %d ].",
+	} else if result.TotalChunks < numchunks {
+		m.log.Warnf("The total chunk size for existing rolling dataset [ %s ] was set to [ %d ] and is being increased to [ %d ].",
+			db, result.TotalChunks, numchunks)
+
+		// if the total chunks increaase grow cid_list while preserving original entries
+		for i := result.TotalChunks; i < numchunks; i++ {
+			update["cid_list." + strconv.Itoa(i) + ".set"] = false
+		}
+	} else if result.TotalChunks > numchunks {
```

Comment:
We still don't allow the total chunks to decrease. That's why there is still a TODO comment in this block. But the `setRolling` function in the import command prevents this function from being called with a decrease in total chunks.